infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/MachineNormalizer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker.environment;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+
+import java.util.stream.Collectors;
+import org.eclipse.che.api.core.ValidationException;
+import org.eclipse.che.api.core.model.workspace.config.ServerConfig;
+import org.eclipse.che.api.workspace.server.spi.InternalMachineConfig;
+import org.eclipse.che.workspace.infrastructure.docker.model.DockerContainerConfig;
+
+/**
+ * Utility for putting stuffs to machine
+ *
+ * @author Alexander Garagatyi
+ * @author Alexander Andrienko
+ */
+public class MachineNormalizer {
+
+  private MachineNormalizer() {}
+
+  public static void normalizeMachine(
```

Comment:
Sure, to me it looks like pure utility method. 
I think we should not overuse DI for classes which hardly will have other implementation to switch to. IMO it just makes code overloaded with huge number of constructor parameters for practically nothing.

Reply:
Using of static methods which contain complicated logic makes testing so hard. We aren't able to stub invocation of static methods and you should prepare all that is needed for method execution without exception in each test where this method is used. So, you should prepare objects in tests for `DockerfileInternalEnvironmentFactory`, `DockerimageInternalEnvironmentFactory`.

Reply:
well, it is always a matter of compromise...there are no black/white decision
I am answering on question :)

for me, clear structure and readability prevales over being able to use the favorite testing framework and I would disagree that mocking always mean easier testing code.

Reply:
I would say it's not about using a favorite testing framework, it's the way of having extensibility/freedom.

If the only worry is about constructor parameters, you can use field injection or setter injection to not overload constructor parameters as well. (Using @InjectMethod is handling that when testing)

Reply:
Do you have an example of what kind of extensibility we can expect from the method under discussion? (for my understanding, maybe I underestimate its value, that is not my code inside)

Reply:
well, you may not know how your code will be used as part of an open source. People may want to extend some classes, replace some logical stuff, etc. 
So, following extensibility model (class instead of static code) you ease the derivative work that may be done
Sometimes it's surprising to see what people do as workaround only because upstream project is not so extensible

Reply:
I do not think it is a good place to discuss it in general, this particular case shows that there are nothing to extend normally.

Anyway I removed this class at all, reworking a bit Docker/Compose config instead